lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1116,64 +1145,99 @@ internal class EvaluatingCompiler(
         }
     }
 
-    private fun compileTyped(expr: Typed): ThunkEnv {
-        val (op, exp, dataType, metas: MetaContainer) = expr
-        val expThunk = compileExprNode(exp)
-        val typedOpParameter = dataType.toAstType().toTypedOpParameter(customTypedOpParameters)
+    /**
+     * Returns a lambda that implements the `IS` operator type check according to the current
+     * [TypedOpBehavior].
+     */
+    private fun makeIsCheck(
+        staticType: SingleType,
+        typedOpParameter: TypedOpParameter,
+        metas: MetaContainer
+    ): (ExprValue) -> Boolean {
+        val exprValueType = staticType.runtimeType
 
-        when (typedOpParameter.staticType) {
-            is SingleType -> {
-            }
-            is AnyType -> {
-                // return trivial results for operations against ANY
-                return when (op) {
-                    TypedOp.CAST -> expThunk
-                    TypedOp.CAN_CAST, TypedOp.CAN_LOSSLESS_CAST, TypedOp.IS ->
-                        thunkFactory.thunkEnv(metas) { valueFactory.newBoolean(true) }
-                }
+        // The "simple" type match function only looks at the [ExprValueType] of the [ExprValue]
+        // and invokes the custom [validationThunk] if one exists.
+        val simpleTypeMatchFunc = { expValue: ExprValue ->
+            val isTypeMatch = when (exprValueType) {
+                // MISSING IS NULL and NULL IS MISSING
+                ExprValueType.NULL -> expValue.type.isUnknown
+                else -> expValue.type == exprValueType
             }
+            (isTypeMatch && typedOpParameter.validationThunk?.let { it(expValue) } != false)
         }
 
-        when (compileOptions.typedOpBehavior) {
-            TypedOpBehavior.LEGACY -> {
-                // no validation needed since type parameters are not honored in this mode anyway
-            }
-            TypedOpBehavior.HONOR_PARAMETERS -> {
-                when (typedOpParameter.staticType) {
-                    is FloatType ->
-                        // check if FLOAT has been given an argument--throw exception since we do not honor it.
-                        if (expr.type.args.any()) {
-                            err(
-                                "FLOAT precision parameter is unsupported",
-                                ErrorCode.SEMANTIC_FLOAT_PRECISION_UNSUPPORTED,
-                                errorContextFrom(expr.type.metas),
-                                internal = false)
```

Comment:
Since `Is`, `Can`, `CanCast` and `CanLosslessCast` are separate types and there is no corresponding similar parent PIG node type for `Typed`, we have to rewrite [L1120](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L1120) - [L1235](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L1235) once for each type, if we want to keep the original function structure. I think this is really cumbersome. 

As for the check of `TypedOpBehavior.HONOR_PARAMETERS`, I will add it back in the next revision.